From 3a33fed1a56dd3104a1718c91b55bb42bef215fa Mon Sep 17 00:00:00 2001
From: Joan Torres Lopez <joantolo@redhat.com>
Date: Tue, 15 Jul 2025 11:29:48 +0200
Subject: [PATCH] schema: Add 'restart-enabled' on screensaver schema

Originally restart/shutdown was always disabled at lockscreen.
This new setting allows to enable it if desired.

To keep the original state, the default is still disabled.
---
 schemas/org.gnome.desktop.screensaver.gschema.xml.in | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/schemas/org.gnome.desktop.screensaver.gschema.xml.in b/schemas/org.gnome.desktop.screensaver.gschema.xml.in
index 5c2db2a..4cf7d3a 100644
--- a/schemas/org.gnome.desktop.screensaver.gschema.xml.in
+++ b/schemas/org.gnome.desktop.screensaver.gschema.xml.in
@@ -1,6 +1,13 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <schemalist gettext-domain="gsettings-desktop-schemas">
   <schema id="org.gnome.desktop.screensaver" path="/org/gnome/desktop/screensaver/">
+    <key name="restart-enabled" type="b">
+      <default>false</default>
+      <summary>Enable restart/shutdown</summary>
+      <description>
+        Allow shutdown and restart actions from the lock screen.
+      </description>
+    </key>
     <key name="idle-activation-enabled" type="b">
       <default>true</default>
       <summary>Activate when idle</summary>
-- 
2.49.0

